src/integrations/xiaomi-mi/xiaomi-mi.service.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  Injectable,
+  Logger,
+  OnApplicationBootstrap,
+  OnModuleInit
+} from '@nestjs/common';
+import noble, { Peripheral, Advertisement } from '@abandonware/noble';
+import { EntitiesService } from '../../entities/entities.service';
+import { ConfigService } from '../../config/config.service';
+import { XiaomiMiConfig } from './xiaomi-mi.config';
+import { makeId } from '../../util/id';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import {
+  SERVICE_DATA_UUID,
+  ServiceData,
+  Parser,
+  Event,
+  EventTypes
+} from './parser';
+import { Sensor } from '../../entities/sensor';
+import { EntityCustomization } from '../../entities/entity-customization.interface';
+import { SensorConfig } from '../home-assistant/sensor-config';
+
+class SensorSet {
+  private readonly name: string;
+  private readonly entitiesService: EntitiesService;
+  private measures: { [kind: string]: Sensor } = {};
+
+  constructor(name: string, entitiesService: EntitiesService) {
+    this.name = name;
+    this.entitiesService = entitiesService;
+  }
+
+  recordMeasure(kind: string, units: string, state: number | string) {
+    let sensor = this.measures[kind];
+    if (!sensor) {
+      const customizations: Array<EntityCustomization<any>> = [
+        {
+          for: SensorConfig,
+          overrides: {
+            deviceClass: kind,
+            unitOfMeasurement: units
+          }
+        }
+      ];
+      const name = `${this.name} ${kind}`;
+      const id = makeId(name);
+      sensor = this.entitiesService.add(
+        new Sensor(id, name),
+        customizations
+      ) as Sensor;
+      this.measures[kind] = sensor;
+    }
+    sensor.state = state;
+  }
+}
+
+@Injectable()
+export class XiaomiMiService implements OnModuleInit, OnApplicationBootstrap {
+  private readonly config: XiaomiMiConfig;
+  private readonly logger: Logger;
+  private sensors: { [address: string]: SensorSet } = {};
+
+  constructor(
+    private readonly entitiesService: EntitiesService,
+    private readonly configService: ConfigService,
+    private readonly schedulerRegistry: SchedulerRegistry
```

Comment:
`schedulerRegistry` doesn't seem to be used, so it can be removed

Reply:
Done